testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -560,6 +562,18 @@ public void testAnalyzeAccessControl()
         assertAccessDenied("ANALYZE nation", "Cannot select from columns \\[.*nationkey.*] in table or view .*.nation", privilege("nation.nationkey", SELECT_COLUMN));
     }
 
+    @Test
+    public void testMetadataFilterColumns()
```

Comment:
How about testing views and materialized views as well?

Reply:
let's do this as a follow-up.